airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -182,7 +180,109 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
         diff = params - schema_path
         assert diff == set(), f"Specified oauth fields are missed from spec schema: {diff}"
 
-    def test_additional_properties_is_true(self, actual_connector_spec):
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_field_declaration(self, spec_diff: DeepDiff):
+        """Check if a 'required' field was added to the spec."""
+        added_required_fields = [
+            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_fields) == 0, f"The current spec has a new required field: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_property(self, spec_diff: DeepDiff):
+        """Check if a a new property was added to the 'required' field."""
+        added_required_properties = [
+            addition for addition in spec_diff.get("iterable_item_added", []) if addition.up.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_properties) == 0, f"The current spec has a new required property: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_field_type_changed(self, spec_diff: DeepDiff):
+        """Check if the current spec is changing the types of properties."""
+        # TODO alafanechere: Do we have allowed non-breaking type transitions?
+        # e.g. Does changing a field type from int to string is breaking the spec?
+        # I assumed it does
```

Comment:
I think in the case of specs a type change is always breaking. If we expect the config to have a value sent as a string but we get an int instead,  the code would probably be doing things on that string that could break, e.g. `value.replace("-", "/")` wouldn't work if the value we got was an integer instead.

Reply:
477a50be86bedbb184e3c927349eddfe2e95f777